lib/install/management/dispatcher.go

```diff
@@ -125,12 +139,108 @@ func (d *Dispatcher) initDiagnosticLogs(conf *metadata.VirtualContainerHostConfi
 	}
 }
 
+func (d *Dispatcher) registerExtension(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infoln("Registering VCH as a vSphere extension")
+	extensionManager := object.NewExtensionManager(d.session.Vim25())
+	extension := types.Extension{
+		Description: &types.Description{
+			Label:   "VIC",
+			Summary: "vSphere Integrated Containers Virtual Container Host",
+		},
+		Company: "VMware, Inc.",
+		Version: "0.2",
+		Key:     "com.vmware.vic",
+	}
+
+	// anonymous function generates a keypair
+	var err error
+	if conf.ExtensionCert, conf.ExtensionKey, err = func() (string, string, error) {
```

Comment:
Yeah, it's pretty much a copypasta -- I didn't just call that code for two reasons but I'm obviously open to input on this:
1) We don't write the keypair out to a file here, so the routine in certificate would have to be refactored somewhat as it doesn't fit the use-case exactly and
2) I'd need to break it into its own package in order to use it from here and from `create` or there's a cyclical import. I didn't want to waste time belaboring it so I copied over the part of the routine that I needed (stripping out the write-to-file parts), wrapped it in an anonymous function, and called it a day.